Wikipedia/Code/Article as a Living Document/ActivityIndicatorCollectionViewFooter.swift

```diff
@@ -0,0 +1,33 @@
+class ActivityIndicatorCollectionViewFooter: UICollectionReusableView {
+    private let loadingIndicator = UIActivityIndicatorView(style: .gray)
+    private let loadingLabel = UILabel()
+
+    override init(frame: CGRect) {
+        super.init(frame: .zero)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupView() {
+        loadingLabel.font = UIFont.wmf_font(.footnote, compatibleWithTraitCollection: traitCollection)
+        loadingLabel.text = WMFLocalizedString("loading-indicator-text", value: "Loading", comment: "Text shown underneath loading indicator.").uppercased()
+
+        let stackView = UIStackView(arrangedSubviews: [loadingIndicator, loadingLabel])
+        stackView.axis = .vertical
+        stackView.spacing = 4.0
+
+        wmf_addSubview(stackView, withConstraintsToEdgesWithInsets: UIEdgeInsets(top: 30, left: 30, bottom: 30, right: 30))
```

Comment:
@mcleinman I actually didn't think we could mix and match auto layout cells with our manually laid out ones, so you made it farther than I thought. I would keep going with this, I think all you need to do is replace this line with something like:

```
        stackView.alignment = .center
        addSubview(stackView)
        stackView.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([ stackView.centerXAnchor.constraint(equalTo: self.centerXAnchor),
                                      stackView.topAnchor.constraint(equalTo: self.topAnchor, constant: 30),
                                      stackView.bottomAnchor.constraint(equalTo: self.safeAreaLayoutGuide.bottomAnchor, constant: 30)])
```

Because `wmf_addSubview` doesn't add any centering constraints. You probably still need some more tweaks, like maybe add >= 30 constraints for leading and trailing in case the screen gets really small and the type is large. Also my `safeAreaLayoutGuide` in the snippet isn't working quite right:

![Screen Shot 2020-10-30 at 10 22 30 AM](https://user-images.githubusercontent.com/3620196/97723555-eb019500-1a99-11eb-96da-55cb263846c7.png)
 
If you were to hook into our manual system, you would subclass `SizeThatFitsReusableView` and override sizeThatFits to position the indicator and label (see `ArticleAsLivingDocSectionHeaderView`. But I don't think you need to do that here.

Reply:
I mistakenly assumed the stack view would take full width, had some end-of-day brain spasm. Thanks, check it out now - should be working well.